modal/_serialization.py

```diff
@@ -470,10 +472,23 @@ def deserialize_params(serialized_params: bytes, function_def: api_pb2.Function,
         api_pb2.ClassParameterInfo.PARAM_SERIALIZATION_FORMAT_PICKLE,
     ):
         # legacy serialization format - pickle of `(args, kwargs)` w/ support for modal object arguments
-        param_args, param_kwargs = deserialize(serialized_params, _client)
+        try:
+            param_args, param_kwargs = deserialize(serialized_params, _client)
+        except DeserializationError:
+            # Fallback in case of proto -> pickle downgrades, where a pickle function could end
+            # up getting bound to proto argument during the downgrade transition.
```

Comment:
what does it mean for a "pickle function [to get] bound to proto argument"?

Reply:
a function that has parameter_info.serialization_format set to pickle getting a `serialized_params` payload in FunctionBindParams that's proto encoded

Reply:
Mostly just asking for clearer comments so we understand the point of this in a few months.